Yoshi/QAKit/EnvironmentMenu/YoshiEncodableEnvironment.swift

```diff
@@ -0,0 +1,25 @@
+//
+//  YoshiEncodableEnvironment.swift
+//  Yoshi
+//
+//  Created by Kanglei Fang on 25/09/2017.
+//  Copyright © 2017 Prolific Interactive. All rights reserved.
+//
+
+import Foundation
+
+/// Encodable and Codable HTTP environment.
+internal struct YoshiEncodableEnvironment: Environment, Codable {
```

Comment:
Why not just make `YoshiEnvironment` `Codable`? Do we really a separate struct just to conform to `Codable`?

Reply:
User might want a different way to encode the choices, like privacy concern.

Reply:
Yeah but if we change the initializer for the environment manager to take in an array of `Environment & Codable` like we talked about, there would be no reason or occasion to directly use `YoshiEnvironment`.